test/clj_momo/lib/es/document_test.clj

```diff
@@ -288,11 +317,58 @@
     (is (= 3 (es-doc/count-docs conn "test_index" "test_mapping" {:ids {:values (range 3)}})))
     (es-index/delete! conn "test_index")))
 
-(deftest ^:integration query-test
+(deftest ^:integration scroll-test
   (let [sample-docs (mapv #(assoc {:_index "test_index"
                                    :_type "test_mapping"
                                    :foo :bar}
                                   :_id (str %))
+                          (range 100))
+        conn (es-conn/connect (th/get-es-config))
+        _ (es-index/delete! conn "test_index")
+        _ (es-index/create! conn "test_index" {})
+        _ (es-doc/bulk-create-doc conn sample-docs "true")
+        {batch-1 :data
+         {scroll-id-1 :scroll_id} :paging} (es-doc/query conn
+                                                         "test_index"
+                                                         "test_mapping"
+                                                         {:match_all {}}
+                                                         {:scroll "10s" :limit 40})
+       {batch-2 :data
+        {scroll-id-2 :scroll_id} :paging} (es-doc/scroll conn
+                                                         scroll-id-1
+                                                         {:scroll "10s"
+                                                          :full-hits? true})
+        {batch-3 :data
+         {scroll-id-3 :scroll_id} :paging} (es-doc/scroll conn
+                                                          scroll-id-2
+                                                          {:scroll "10s"})
+
+        {batch-4 :data
+         {scroll-id-4 :scroll_id} :paging} (es-doc/scroll conn
+                                                          scroll-id-3
+                                                          {})]
+    (is (= scroll-id-1
+           scroll-id-2
+           scroll-id-3
+           scroll-id-4))
```

Comment:
I do not understand this test, from the doc I would expect all of them to be different.

https://www.elastic.co/guide/en/elasticsearch/reference/5.1/search-request-scroll.html

The initial search request and each subsequent scroll request returns a new _scroll_id — only the most recent _scroll_id should be used.

Reply:
ES may send you a new one when the scroll id is about to expire, but it will send you the same one if it is not.
I have no clue on how know how to simulate that.
If it expires before the next scroll request (with a sleep between two calls for instance), the next scroll request will return empty data and empty scroll_id because ES would have deleted the corresponding search context.

Reply:
if I understand correctly you are given a new scroll_id when it expires?

Also if this is the case then I guess that to be clean we must be able to clear the scroll context with 

https://www.elastic.co/guide/en/elasticsearch/reference/5.5/search-request-scroll.html#_clear_scroll_api

So I think you should;

1. use a shorted lifetime for the scroll_id, like 1s
2. make 2 requests in less than 1s
3. wait >1s
4. make other requests
5. check the scroll_id is different but still work
6. delete the scroll_id
7. make another request with the scroll_id and check it is refused

Also I believe you should add that delete scroll_id in the API.